app/registries/foreman/access_control.rb

```diff
@@ -22,7 +22,13 @@ def map
         mapper = Mapper.new
         yield mapper
         @permissions ||= []
-        @permissions += mapper.mapped_permissions
+        mapper.mapped_permissions.each do |permission|
+          if (perm = permission(permission.name))
+            perm.actions.concat(permission.actions)
```

Comment:
I think if you just add an warning here, we can add the proper API in another PR and consider this a fix for the bug :)